### PR TITLE
Restructuring

### PR DESCRIPTION
**Pull request by petterill**
- deleted dependency-tracking.md
- deleted js-project-overview
- deleted performance.md
- moved outline to drafts folder
- merged PR by jimppa so that all coding standards live in one doc. This PR merges python, python-ideas, es and kuva confluence stuff into one doc.
- moved technology stuff from coding standards to technology-choices
- removed duplicate lines from coding standards
- added gitignore
- improved readme

With the developer site implemented, the structure would be as follows:
- Front page / Readme
- APIs
- Developments practices
  - Coding standards
  - Technology choices
  - Django configuration
  - Maps
  - Version control
  - Accepting contributions
  - Scrum project management
  - Drafts
    - Best practices outline

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # Best practices for developing for City of Helsinki
 
-This happy little repo aims to be place for the practical details for developing code for City of Helsinki. Currently it is a bit outdated, but we hope to fill in stuff such as how to do authentication and configuration.
+This documentation covers the practical details for developing open source code for the City of Helsinki. The target audience is software development consultants working in Helsinki projects as well as Helsinki software development staff. This documentation is produced jointly by the software developers of the Executive Office and the citys divisions. Not all practices stated here will apply to all projects within the city but these are intended as general outlines of best practices. If you wish to contribute, please make a pull request in the [GitHub repository](https://github.com/City-of-Helsinki/bestpractice).
```
Comment (helsinkisolution-integration) [inline README.md]: - also, people making contributions to City of Helsinki software (see CONTRIBUTING.md in other repositories)
- contribution guidelines could be put into a separate CONTRIBUTING.md (template at https://helsinkisolutionoffice.atlassian.net/wiki/spaces/DD/pages/753772/Documentation+practices)

Comment (helsinkisolution-integration) [inline README.md]:
```suggestion
This documentation covers the practical details for developing open source code for the City of Helsinki. The target audience is software development consultants working for Helsinki codebases, Helsinki software development staff, and people offering contributions to Helsinki software. This documentation is produced jointly by the software developers of the Executive Office and the divisions of the city. Not all practices stated here will apply to all projects within the city but these are intended as general outlines of best practices. If you wish to contribute, please make a pull request in the [GitHub repository](https://github.com/City-of-Helsinki/bestpractice).
```

File: coding-standards.md [hunk 2]
```diff
@@ -0,0 +1,287 @@
+# Coding standards
+
+## Backend code standards
+
+### Python conventions
+
+- Newer projects use [black](https://github.com/ambv/black) for Python code formatting. We follow the basic config, without any modifications. It should be required for new projects and is warmly welcome for the old ones.
+- We also use [isort](https://github.com/timothycrosley/isort) for keeping all the imports nice and clean. To make it compatible with black add the following line either to your project's setup.cfg file's `[isort]` section or to isort.cfg: bash multi_line_output=3 include_trailing_comma=True force_grid_wrap=0 use_parentheses=True line_length=88
```
Comment (helsinkisolution-integration) [inline coding-standards.md]: Literal formatting to isort.cfg contents

File: coding-standards.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+    and .travis.yml files,
+    create a directory for tests (named `tests`),
+    *but* make a top level directories for the actual package code
+    named `src`, *not* [the same as the package](https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure).
+    In case the project is not a reusable part of a bigger application,
+    but a standalone app requiring a more complex installation
+    (like a Django project requiring a database setup etc.),
+    you can skip the setup.py setup.cfg.
+
+*   Use static typing for APIs instead of comments.
```
Comment (helsinkisolution-integration) [inline coding-standards.md]:
```suggestion
*   Use static typing for functions instead of comments.
```

File: dependency-tracking.md [hunk 4]
```diff
@@ -1,107 +0,0 @@
...
-
-Nice to have
-- cli tool for dev/CI use (checking before pushing, seeing status in PRs)
-- Nice report of all repos
-- Configuration without clicking a lot in GUI
-  (e.g. to change the email address for notifications)
-- OSS
-
-# Cloud dependency checkers
-
```
Comment (helsinkisolution-integration) [inline dependency-tracking.md:15]: While I can clearly see that this file is of no use, making a recommendation of using dependency checkers (and which) should be added to dev site's backlog.

File: es.md [hunk 5]
```diff
@@ -1,88 +0,0 @@
-### ECMAScript/JS conventions
-
-*   ES7 (with babel and polyfills), not Flow or TypeScript
-*   eslint with airbnb style guide
-*   yarn and webpack, not bower or grunt
-    *   Additional webpack modules need to go through architecture review
```
Comment (helsinkisolution-integration) [inline es.md:6]: this recommendation is missing from coding standards

File: es.md [hunk 6]
```diff
@@ -1,88 +0,0 @@
...
-*   piwik for analytics and Sentry for error tracking
```
Comment (helsinkisolution-integration) [inline es.md:7]: this too

File: es.md [hunk 7]
```diff
@@ -1,88 +0,0 @@
...
-*   travis for CI
-*   Use browserstack/saucelabs or similar to test mainstream browsers and devices
-*   Create a visual style guide with the UI components
-    *   Test the components visually with gemini
-    *   A single page with all the components for human consumption,
-      separate pages with component per page for CI
-*   process.env can be read only from one file,
-    separate ENVs that are read only build time and those that are read runtime
-*   Avoid using target="\_blank", and if you do, use noreferrer and noopener
```
Comment (helsinkisolution-integration) [inline es.md:16]: these should be checked if they are still relevant.  I lack the expertise.

File: es.md [hunk 8]
```diff
@@ -1,88 +0,0 @@
...
-*   Do not include node\_modules in source.
-    If you need to build something custom, built it properly so that any (sub)dependencies
-    do not get locked.
-    If you need to patch upstream versions, make PRs upstream and document the related PR to
-    the repo so that the hacked version can be changed back to the official one later.
-*   [Use await/async instead of Promises](https://mathiasbynens.be/notes/async-stack-traces).
```
Comment (helsinkisolution-integration) [inline es.md:22]: These are still relevant, I guess?  And not in coding standards.

File: es.md [hunk 9]
```diff
@@ -1,88 +0,0 @@
...
-    And especially always catch the errors, as otherwise they will go to a black hole and you'll never hear from them (at least write a `.catch(console.log.bind(console))`).
-
-
-#### React
-
-*   react-intl for translations
-*   redux as the store
-*   redux-actions
-*   redux-promise, saga or thunk (needs comments)
-*   Separate API HTTP calls to a separate module/file
```
Comment (helsinkisolution-integration) [inline es.md:80]: This is probably still relevant and I think not found in coding-standards.md

File: performance.md [hunk 10]
```diff
@@ -1,12 +0,0 @@
...
-  * run against every new commit / PR
-  * note that lighthouse will require some unknown permissions? for running. Maybe safe?
-* backend tracking
-  * static requests
-* storage of
-  * SaaS? (new relic, datadog)
-  * Whisper (no metadata)
-* alerts
-  * time threshold
-
```
Comment (helsinkisolution-integration) [inline performance.md:12]: We should probably ticketise specifying how performance should be monitored

File: python-ideas.md [hunk 11]
```diff
@@ -1,55 +0,0 @@
...
-* devpi for faster rebuilds
-* Specifying Django settings
-    * /etc, dotfile in user's home
-    * ENV
-    * command line options
-* even Django projects should be testable
-* codecov, not coveralls
-* Django database settings should have timeout to handle Kubernets services being down
-* check that imagemagick isn't used
-* Run tests with deprecation warnings on
```
Comment (helsinkisolution-integration) [inline python-ideas.md:29]: Here are also some good ideas.  Doesn't anyone have time to write them into coding standards?

**Review by helsinkisolution-integration — COMMENTED** [review]
The restructuring itself is good.  The content is problematic: I feel that a lot of stuff is being thrown out just because no one wants to update it.  For instance, does this mean that we *don't* expect yarn anymore and devs should use whichever build/packaging tool they want?

File: es.md (hunk 9 above)
Comment (petterill) [reply es.md:80]: added

File: es.md (hunk 8 above)
Comment (petterill) [reply es.md:22]: added

File: es.md (hunk 7 above)
Comment (petterill) [reply es.md:16]: I'll create an issue for this

File: es.md (hunk 5 above)
Comment (petterill) [reply es.md:6]: added

File: es.md (hunk 6 above)
Comment (petterill) [reply es.md:7]: added, exept for the piwik and sentry stuff, should they be in coding standard or some other doc?

File: python-ideas.md (hunk 11 above)
Comment (petterill) [reply python-ideas.md:29]: added these

File: performance.md (hunk 10 above)
Comment (petterill) [reply performance.md:12]: Created the issue DEV-10 for this

File: dependency-tracking.md (hunk 4 above)
Comment (petterill) [reply dependency-tracking.md:15]: Created the issue DEV-11 for this

**Review by pkalliok — APPROVED** [review]
wow!  Thanks @petterill

**Pull request merged**